### ISSUE TITLE
Bugs with Pick up and haul & PUAH+

### ISSUE THREAD

**Issue by RdLeaf** [jptrrs/HumanResources#174]
Hello, it seems i found a bug with pick up and haul. this problem happens whenever I have to move the books and reinstall the bookshelf. I've been able to recreate the problem in dev mode, but only when the circumstances were right.

First, there needed to be a stockpile zone present (pawns will haul books one by one otherwise), then have a pawn add in all the books. Deconstruct the bookshelf and create a new one, have a pawn add in all the books to the new bookshelf. This should then give you an error log. However, even then the error log will not always appear, and you have to try again.

HugsLIB log: https://gist.github.com/HugsLibRecordKeeper/3a4b6485f9b504c2e242ce076cabd075

As you can see I tried this in multiple new games to make sure. (There were more tries but I forgot to upload.)

**Comment by RdLeaf** [comment]
I've also had the dissappearing  bookshelf bug too, but I couldn't pin point problem.

**Comment by modulozero** [comment]
See also: https://github.com/jptrrs/HumanResources/issues/165 :)

**Comment by jptrrs** [comment]
RdLeaf: could you try this and tell me if that solves it?
https://github.com/jptrrs/HumanResources/issues/165#issuecomment-1059604819

**Comment by RdLeaf** [comment]
Sadly it didn't. https://gist.github.com/HugsLibRecordKeeper/3efb663fc7a457ce6c5f4ebfead8307a

**Comment by jptrrs** [comment]
> Sadly it didn't. https://gist.github.com/HugsLibRecordKeeper/3efb663fc7a457ce6c5f4ebfead8307a

It was worth the shot. But knowing this, I set out to try and reproduce the issue. It took more than two attempts, but I managed. And thanks to your logs, I'm confident to say it happens when the pawn's inventory includes anything that's not a book, something that can happen with PUAH.  Deconstructing one shelf and forcing your pawns to relocate the books to another one triggered it because of the dropped wood from the shelf, pawns would typically grab that along the books. Hence, you got: "driver.job=(UnloadYourHauledInventory (Job_1098) A=Thing_WoodLog2711 B=Thing_BookShelf2709".
The fix is simple. I'll be including it with the next update.
Thanks!
